cloudflare/resource_cloudflare_healthcheck_test.go

```diff
@@ -0,0 +1,178 @@
+package cloudflare
+
+import (
+	"fmt"
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccCloudflareHealthcheckTCPExists(t *testing.T) {
+	t.Parallel()
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+
+	rnd := generateRandomResourceName()
+	name := fmt.Sprintf("cloudflare_healthcheck.%s", rnd)
+	var healthcheck cloudflare.Healthcheck
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckCloudflareHealthcheckTCP(zoneID, "tcp-test-healthcheck", rnd),
```

Comment:
We can sub out `"tcp-test-healthcheck"` with `rnd` here to ensure the name is always unique. In the cases of updated, I generally do something like `rnd + "-updated"` to be able to track the differences if that is something your tests need.

Reply:
Updated accordingly.